### PR TITLE
FIX: Rollback granularity on fetcher sources to daily

### PR DESCRIPTION
**Pull request by lianga888**
For every fetcher source, take the last signal on a given day for
a given asset or data source as the signal for the entire day

### CODE REVIEW

File: zipline/data/data_portal.py [hunk 1]
```diff
@@ -190,18 +187,21 @@ def handle_extra_source(self, source_df, sim_params):
         for group_name in group_names:
             group_dict[group_name] = grouped_by_sid.get_group(group_name)
 
+        extra_source_df = None
```
Comment (jbredeche) [inline zipline/data/data_portal.py]: can you just directly initialize `extra_source_df = pd.DataFrame()` here, then you don't need the None check below?

File: zipline/data/data_portal.py [hunk 2]
```diff
@@ -1360,24 +1363,13 @@ def get_fetcher_assets(self, day):
...
-        asset_list = self._extra_source_df.loc[date_to_use]["sid"]
+        day = normalize_date(dt)
 
-        # make sure they're actually assets
-        asset_list = [asset for asset in asset_list
-                      if isinstance(asset, Asset)]
-
-        return asset_list
+        assets = self._extra_source_df.loc[day]['sid']
+        if isinstance(assets, pd.Series):
```
Comment (jbredeche) [inline zipline/data/data_portal.py]: when would they not be `pd.Series`?

Comment (lianga888) [inline zipline/data/data_portal.py]: I'm not that familiar with pandas, but I noticed that if `self._extra_source_df` only has 1 row with the indexed `day`, then `self._extra_source_df.loc[day]` returns a Series and then `self._extra_source_df.loc[day]['sid']` returns a scalar. Only if there are multiple rows with that `day` does `self._extra_source_df.loc[day]['sid']` return a Series

File: zipline/data/data_portal.py (hunk 1 above)
Comment (lianga888) [inline zipline/data/data_portal.py]: yeah, that makes sense. Also, added a check in `get_fetcher_assets` to return `[]` when the df is empty

File: zipline/data/data_portal.py [hunk 3]
```diff
@@ -139,6 +139,9 @@ def __init__(self,
         }
         self._equity_daily_reader_array_data = {}
 
+    def _reindex_extra_source(self, df, source_date_index):
```
Comment (jbredeche) [inline zipline/data/data_portal.py]: why did you move this to be a separate method?  (it's not wrong, I'm just curious).  it took me a few reads to realize that it's not a functional change.

File: zipline/data/data_portal.py [hunk 4]
```diff
@@ -190,18 +187,21 @@ def handle_extra_source(self, source_df, sim_params):
         for group_name in group_names:
             group_dict[group_name] = grouped_by_sid.get_group(group_name)
 
+        extra_source_df = pd.DataFrame()
```
Comment (jbredeche) [inline zipline/data/data_portal.py]: can you add a comment explaining what `extra_source_df` is here?  looks like a dataframe to which we're appending other dataframes below?

File: zipline/data/data_portal.py (hunk 3 above)
Comment (lianga888) [inline zipline/data/data_portal.py]: It gets overridden in live_data_portal, which does not forward fill https://github.com/quantopian/qexec/pull/9440
